openedx_events/enterprise/signals.py

```diff
@@ -24,11 +24,59 @@
 
 # .. event_type: org.openedx.enterprise.subsidy.redemption-reversed.v1
 # .. event_name: SUBSIDY_REDEMPTION_REVERSED
-# .. event_description: emitted when an enterprise subsidy is reversed.
+# .. event_description: (deprecated) emitted when an enterprise subsidy is reversed.
 # .. event_data: SubsidyRedemption
 SUBSIDY_REDEMPTION_REVERSED = OpenEdxPublicSignal(
     event_type="org.openedx.enterprise.subsidy.redemption-reversed.v1",
     data={
         "redemption": SubsidyRedemption,
     }
 )
+
+
+# .. event_type: org.openedx.enterprise_subsidies.ledger_transaction.created.v1
+# .. event_name: LEDGER_TRANSACTION_CREATED
+# .. event_description: emitted when an enterprise ledger transaction is created.
+# .. event_data: LedgerTransaction
+LEDGER_TRANSACTION_CREATED = OpenEdxPublicSignal(
```

Comment:
What is ledger? Is this a technology used widely by the community? Could you include some docs?

Reply:
Also, it'd be helpful to include how the event will be used.

Reply:
The best doc to start with is probably this: https://github.com/openedx/enterprise-access/blob/main/docs/decisions/0004-add-access-policy-functionality.rst#context

A `Ledger` is a model defined in the openedx-ledger repo, it's a core model used in the context of edX Enterprise. https://github.com/openedx/openedx-ledger/blob/main/docs/decisions/0002-ledger-balance-enforcement.rst

It's installed only into enterprise-subsidy AFAIK: https://github.com/openedx/enterprise-subsidy/tree/main/docs/decisions

Reply:
I don't ledgers (or enterprise-subsidy,access)to be used widely by the community - my guess is that 2u is the only entity currently using it.

Reply:
FWIW, the events will be used to really state transitions from enterprise-subsidy back to enterprise-access so that a related record, `LearnerContentAssignment`, can update its state based on the new state of its related transaction.
https://github.com/openedx/enterprise-access/blob/main/docs/decisions/0012-assignment-based-policies.rst
https://github.com/openedx/enterprise-access/blob/430d768ee4c6355b353e22f5b0f7902d169850bc/enterprise_access/apps/content_assignments/models.py#L234

Reply:
Thank you. Could you include all those docs in the PR cover letter? I'd be useful for future adopters.

Reply:
Since this is not widely used or known, can we include links to the ledge documentation in the in-line annotations?